pkg/store/bucket.go

```diff
@@ -1154,16 +1160,14 @@ func chunksSize(chks []storepb.AggrChunk) (size int) {
 
 // LabelNames implements the storepb.StoreServer interface.
 func (s *BucketStore) LabelNames(ctx context.Context, req *storepb.LabelNamesRequest) (*storepb.LabelNamesResponse, error) {
-	resHints := &hintspb.LabelNamesResponseHints{}
-
-	g, gctx := errgroup.WithContext(ctx)
+	reqSeriesMatchers, err := storepb.MatchersToPromMatchers(req.Matchers...)
+	if err != nil {
+		return nil, status.Error(codes.InvalidArgument, errors.Wrap(err, "translate request labels matchers").Error())
+	}
 
-	s.mtx.RLock()
```

Comment:
Note that `s.mtx.RLock()` is now only called after checking hints. Previously wrong hints would return error without unlocking this lock.